### ISSUE TITLE
GPL is too restrictive

### ISSUE THREAD

**Issue by DarkDrek** [piwik/piwik-dotnet-tracker#23]
Using this library would force me to provide the source code of my application as well.
Could you provide this library under the same license as the official PHP Tracking API? [Link](https://github.com/piwik/piwik-php-tracker/blob/master/LICENSE)

**Comment by julienmoumne** [comment]
This project should definitely have the same license as the php one. Thanks for spotting this.

**Comment by mattab** [comment]
Consider the project is under MIT or BSD license and you can freely use & redistribute it.  we'll change the license as soon as possible, cc @julienmoumne

**Comment by reinholdaudials** [comment]
+1

**Comment by julienmoumne** [comment]
We actually need to contact all the contributors to ask for their permission.

Is there a particular process by which we should collect their consent or would a simple e-mail suffice?

**Also, do note the GPL restriction you mentionned only applies if you propagate and convey your code.**

If you do not propagate your code with this library, the restriction does not apply. This is a common misconception.

See [Basic Permissions](https://www.gnu.org/licenses/gpl-3.0.en.html#section2).

**Comment by mattab** [comment]
we made a little mistake when picking the GPL license for this SDK in the early days (and I missed this fact during early review), as the GPL license for a SDK prevents the SDK users from bundling their app along with the SDK and sell it/distribute it to others. (or their app would have to be GPL too since it's linked to the SDK and GPL applies). I consider it a simple mistake that we can easily correct by changing license to similar license as Java/PHP/iOS/Android SDKs :+1:  (I'd say we can change the license without hearing back from all other 3 contributors since you contributed almost everything and other contribs are minor..)

**Comment by opcon** [comment]
I agree with changing the license from GPL, as @mattab said it means I can't legally distribute my game using Piwik Analytics.

**Comment by julienmoumne** [comment]
I will ask contributors to give their consent by commenting in this issue. 

In the mean time, as mentioned in my previous comment, if you are using the library on your own servers you are fine.

**Comment by mattab** [comment]
Awesome thanks @julienmoumne :+1:

**Comment by robocoder** [comment]
Yes, IMO the GPL for the dotnet tracking library would encumber any app that is distributed with it. The LGPL might be a good middle ground as it would require distributors to open source their changes to the library. But whether you choose to relicense as BSD, MIT, or some other liberal license, you will need the approval of its contributors.

**Comment by robocoder** [comment]
For relicensing, getting the approval of contributors is required because the Piwik project doesn't use a formal copyright transfer for contributions.

You don't need approval for trivial contributions, or contributions that are no longer in the current code base.

**Comment by tetsuo13** [comment]
👍

**Comment by fushnisoft** [comment]
Fine with me! 👍

**Comment by evilham** [comment]
**TL;DR:** Make it clear which license the project is changing to! Don't use PHP License but MIT or BSD instead.

While I'm glad to see someone already brought this up and developers seem to agree there is a licensing issue, I don't see it moving forward in a nice direction, so my two cents (btw: IANAL!):

Some contributors are "agreeing" to the general idea of changing the license away from GPL, that's nice, but it is not **clear** what they are agreeing to change it to.

From reading the thread, I _think_ the current proposal is changing to PHP license, which I don't think is such a good choice: [PHP License](http://www.php.net/license/3_01.txt) is way too PHP specific. Article 4 even says: 

> software may not be called "PHP", nor may "PHP" appear in their name,

(Which, for example, means `RALPHPHOTO` would not be an allowed name)

In any case, the PHP License is over complicated for this library, as mentioned by others MIT or BSD would be a better fit. As an example, [piwik-php-tracker](https://github.com/piwik/piwik-php-tracker/blob/master/LICENSE) uses BSD.

**Comment by robocoder** [comment]
Since the "+1's" don't specifically vote in favor of a specific license, we should formalize the proposal. Therefore...

The proposal is to change this project's license to BSD. Re-licensing under a BSD license would be consistent with the other official Piwik trackers. The call is now out for the following contributors to please vote either for/:+1: or against/:-1: the proposal.

|Contributor|
|------------|
| @julienmoumne |
| @tetsuo13  |
| @fushnisoft |
| @opcon |
| @tsteur |
| @ptr1120 |
| @mironych |

**Comment by julienmoumne** [comment]
Which BSD header should we use?

**Comment by julienmoumne** [comment]
Sorry for the state changes. I'm working on the issue and misclicked.

**Comment by julienmoumne** [comment]
fixed in 4883823b5ba5f1a482adb313988ce0456103bc00
